plugins/thunderbolt/fu-thunderbolt-image.c

```diff
@@ -624,14 +697,20 @@ fu_plugin_thunderbolt_validate_image (GBytes  *controller_fw,
 	if (hw_info->id == 0)
 		return UNKNOWN_DEVICE;
 
-	locations = is_host ?
-			    get_host_locations (hw_info->id) :
-			    get_device_locations (hw_info->id);
-	if (locations == NULL) {
-		g_set_error_literal (error,
-				     FWUPD_ERROR, FWUPD_ERROR_NOT_SUPPORTED,
-				     "FW locations to check not found for this controller");
-		return VALIDATION_FAILED;
+	if (is_host) {
+		locations = get_host_locations (hw_info->id);
+		if (locations == NULL) {
+			g_set_error_literal (error,
+					     FWUPD_ERROR, FWUPD_ERROR_NOT_SUPPORTED,
+					     "FW locations to check not found for this controller");
+			return VALIDATION_FAILED;
+		}
+	} else {
+		locations = get_device_locations (hw_info->id, &controller, error);
```

Comment:
So you get the MC DROM entry location based on the controller and use the same location to read it from the image file? This looks wrong. What if the DROM was changed (e.g. a typo in the vendor/model name was fixed)?

Reply:
That's a fair point (even though probably quite unlikely to have device with MC support and that the device has typo in some string field that requires adding more characters to the string). Since `compare_locations()/compare()` only takes one offset we can't use them directly. Then I guess we just need to read both MC values separately using `read_drom_entry_location()` and compare them.

Reply:
Or changing these functions to support separated location info for the image.
I'm not sure which one will be cleaner. Probably your way is better, as it allows you to compare first if the MC entry is found in both the controller and the image (fail if it founds only in one of them) and only then compare them, i.e. something like:
```
if (found_in_controller != found_in_image)
{
    return FALSE;
}
if (found_in_controller)
{
    return value_from_controller == value_from_image;
}
return TRUE;
```

Separating it will also make sure no one will add another location to the table under this one.
With the current implementation, MC entry must be the last in the table (again, worth a comment if it kept as-is) as it can have offset 0, which is used as a sentry for end of the table, see [line 557](https://github.com/hughsie/fwupd/blob/6f6d58e5caaca70b1dc7a8fac0efd9225fdc2f3e/plugins/thunderbolt/fu-thunderbolt-image.c#L557).